### ISSUE TITLE
Supporting New Outlook

### ISSUE THREAD

**Issue by daghemo** [aluxnimm/outlookcaldavsynchronizer#396]
I'm previewing the new Outlook for Window. 
See https://support.microsoft.com/en-au/office/getting-started-with-the-new-outlook-for-windows-656bb8d9-5a60-49b2-a98b-ba7822bc7627.
As far as I understand, this will not support Outlook CalDav Synchronizer, since it is a COM add-in and COM add-ins will not be supported.
Non an issue on itself, but a very sad news. :(

**Comment by Backxtar** [comment]
Would also like to see this in the new outlook.

**Comment by otto-liljalaakso-nt** [comment]
I depend on CalDav Synchronizer for seeing my work schedule in my personal calendar.
This issue has forced me to stop using the New Outlook.
I understand that fixing this is a large and difficult task,
but I hope it can happen before the old Outlook becomes unusable.

**Comment by alfureu** [comment]
Please, cannot live without this plugin. Just wanted to highlight that some organisations are actively enforcing the use of the New Outlook in their premises (no kidding). Would be great to have this plugin working with the new version of Outlook

**Comment by daniel-lerch** [comment]
The new Outlook only supports Web Add-Ins. There is a JavaScript API available to interaction with emails ([API Reference](https://learn.microsoft.com/en-us/javascript/api/requirement-sets/outlook/preview-requirement-set/outlook-requirement-set-preview?view=outlook-js-preview)). However, I cannot find a similar API to access calendar events. There is a calendar API in Microsoft Graph ([API Reference](https://learn.microsoft.com/en-us/graph/api/resources/calendar?view=graph-rest-1.0)) but that only supports Microsoft accounts which is likely not what people are using a CalDAV synchronizer for.

So if didn't miss something, it will not be possible to rewrite CalDAV synchronizer as a Web Add-In as long as Microsoft does not introduce additional JavaScript APIs.

**Comment by aluxnimm** [comment]
Yes @daniel-lerch that's exactly the problem atm. And even if a new api would be available also for use with imap accounts, a complete rewrite of the whole project would be necessary and we don't have the resources for that right now.

**Comment by Serpensin** [comment]
I just found out, that you don't need a synchronizer for Google calendars at least.

**Comment by klodoma** [comment]
> I just found out, that you don't need a synchronizer for Google calendars at least.

What do you mean by that? Can you sync google calendar with the new outlook?

**Comment by Serpensin** [comment]
> @Serpensin
> 
> > I just found out, that you don't need a synchronizer for Google calendars at least.
> 
> What do you mean by that? Can you sync google calendar with the new outlook?

Yes. I just tried it and it syncs both ways.

**Comment by klodoma** [comment]
> > @Serpensin
> > > I just found out, that you don't need a synchronizer for Google calendars at least.
> > 
> > 
> > What do you mean by that? Can you sync google calendar with the new outlook?
> 
> Yes. I just tried it and it syncs both ways.

I didn't find it where to add a google calendar in the new outlook? Can you point that out please. It would be a great help.

**Comment by Serpensin** [comment]
> > > @Serpensin
> > > > I just found out, that you don't need a synchronizer for Google calendars at least.
> > > 
> > > 
> > > What do you mean by that? Can you sync google calendar with the new outlook?
> > 
> > 
> > Yes. I just tried it and it syncs both ways.
> 
> I didn't find it where to add a google calendar in the new outlook? Can you point that out please. It would be a great help.

I can't exactly remember, how I did it. It was ether automatic, or I added the calender as ics link.

**Comment by klodoma** [comment]
> > > > @Serpensin
> > > > > I just found out, that you don't need a synchronizer for Google calendars at least.
> > > > 
> > > > 
> > > > What do you mean by that? Can you sync google calendar with the new outlook?
> > > 
> > > 
> > > Yes. I just tried it and it syncs both ways.
> > 
> > 
> > I didn't find it where to add a google calendar in the new outlook? Can you point that out please. It would be a great help.
> 
> I can't exactly remember, how I did it. It was ether automatic, or I added the calender as ics link.

I see, but AFAIK ICS is read-only mode so you cannot manage your events in Outlook. It's an important detail.

**Comment by Serpensin** [comment]
I know. I don't know, how that works, but it worked for me.

Gesendet von Outlook für Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by mojtaba-yekta** [comment]
What I did was first I synced the old Outlook calendar with my Google Calendar by CalDav Synchronizer. Then I synced my Google Calendar with New Outlook with these to steps:

1. Export Google Calendar: Open Google Calendar on a desktop browser. Click the three-dot button next to the calendar you wish to sync. Click "Settings and sharing" from the pop-up window. Click "Integrate calendar" from the sidebar. Scroll down and copy the URL under "Secret address in iCal format"

2. Import to Outlook: Open up Outlook and press the "Add calendar" option to add an external calendar. Select "Subscribe from the web" from the left-hand menu. Paste the URL you copied from your Google Calendar into the box provided

**Comment by PC-SPEZIALIST** [comment]
Maybe the **New Outlook** will support CalDAV and CardDAV natively in a few years. 😅

**Comment by netzwerch** [comment]
A different kind of solution could be to use an additional tool that syncs two icals outside of Outlook like vdirsyncer or ics2caldav
Outlook currently offers (readonly) .ics calendar publishing, so this this could be a solution for some people.

**Comment by NapoSky** [comment]
just came here from a fresh install of mailcow-dockerized to "test the solution". Quite sad to see that the new outlook doesn't do shit about retrocompatibility... 🥲

**Comment by ballerbude** [comment]
Not having this app in the new Outlook would be a really sad outcome.

**Comment by foorschtbar** [comment]
_FYI - From Microsoft Admin Center:_

Microsoft is updating the opt‑out phase start date for new Outlook for Windows in Enterprise environments from April 2026 to March 2027.

**When this will happen:**

- The opt‑out phase for Enterprise environments will now begin in March 2027 (previously April 2026), providing organizations with 12 months of lead time to prepare. 
- GCC High and DoD timelines will be communicated separately at a later date.

**Why is there a change to the timeline?**

We’re seeing strong and accelerating adoption of new Outlook as organizations progress on timelines that match their readiness. At the same time, we continue to invest heavily in expanding capabilities and addressing feedback from customers who want to go further with new Outlook. To ensure organizations have the time they need to prepare—and to fully realize the value of ongoing innovation—we’re extending the opt-out timeline and providing 12 months of lead time as we continue delivering key features and improvements.

**How this affects your organization:**

**Who is affected:**

- Microsoft 365 administrators managing Outlook for Windows.
- Users of classic Outlook for Windows.
- Organizations that previously disabled the new Outlook toggle or have not yet begun migration planning.

**What will happen:**

- Migrated users will receive in‑app onboarding prompts and can switch back to classic Outlook if needed.
- Existing policies that disable or limit access to new Outlook will remain in effect.
- Admins can continue using Admin‑controlled migration to move users in a staged, reversible manner ahead of time.
- Users will not be migrated if one or more of the following is true:
- You have opted out users using the Policy of automatic migration
  - New Outlook toggle is hidden via policy
  - Perpetual license is in use
  - On premise account
  [...]

**Comment by schindlerchristian** [comment]
I faced the same problem and initially built my own server-side connector to sync my CalDAV calendar with Microsoft 365 via the Graph API. The approach worked well enough that I turned it into a proper product: CalDAVconnect (https://caldavconnect.de/). Currently in public beta and free to use.

Instead of relying on a COM add-in inside Outlook, it syncs your CalDAV server (Nextcloud, SOGo, Radicale, etc.) directly with Microsoft 365 - server-to-server, in real time. It works with New Outlook, Classic Outlook, Outlook Web, and Outlook mobile, since the sync happens at the cloud level rather than the client level.

**Comment by ballerbude** [comment]
@schindlerchristian while an interesting idea, it's unfortunately neither open source nor self-hostable, which is a deal breaker at least for me. 

Nothing wrong with earning some money, but advertising a closed source, proprietary product/service on github may be the wrong approach.

**Comment by schindlerchristian** [comment]
@ballerbude Fair point on both counts. GitHub Issues on an open source project probably isn't the ideal place to mention a commercial service - I should have been more mindful of that.

A self-hosted option - where you run CalDAVconnect on your own infrastructure e.g. via Docker  is something I'm actively considering. It wouldn't make the project open source, but it would address the data sovereignty concern. What do you think about that?

**Comment by foorschtbar** [comment]
And at least for me, I wouldn’t have come here if I’d had access to the M365 Graph API. The Outlook CalDav Synchronizer is my solution for accessing M365 data with a “signed-in” client when I can’t authenticate through an app that needs to be registered in Azure/Entra.

**Comment by schindlerchristian** [comment]
@foorschtbar That's a fair and important distinction. CalDAVconnect uses the Microsoft Graph API and does require an Azure app registration with OAuth - so if that's blocked in your environment, it won't work for you either. The Outlook CalDAV Synchronizer's client-side approach does have advantages in restricted corporate environments that a cloud service simply can't replicate.

**Comment by ballerbude** [comment]
@schindlerchristian that would be a step in the right directio, I guess.

I vibecoded a tool with Claude that works similar way and runs as a docker container. I fortunately have access to Entra ID so I could register an app, and it seems so far to do what I want, with some minor polishing still required. Thanks for the inspiration.

**Comment by schindlerchristian** [comment]
@ballerbude Great to hear you got something working! That's exactly the kind of problem it solves well. If you ever want to compare notes on the CalDAV/Graph API edge cases - recurring events, timezone handling, conflict resolution - feel free to reach out. There are more than a few interesting quirks. 😄

**Comment by DennisGaida** [comment]
The selfhosted and open-source alternative is https://github.com/ridafkih/keeper.sh - haven't tried it out, but this seems to solve many problems.

**Comment by PC-SPEZIALIST** [comment]
Thanks for the nice links to [Keeper.sh](https://www.keeper.sh/) and [CalDAVconnect](https://caldavconnect.de/). 👍
By the way, do you plan [CardDAVconnect](https://carddavconnect.de/) too, @schindlerchristian?

**Comment by schindlerchristian** [comment]
@PC-SPEZIALIST Thanks! CardDAV (contacts sync) is definitely on the roadmap. Right now the focus is on making calendar sync rock-solid, but CardDAV is the logical next step. Stay tuned! 🙂

**Comment by cymumtaz** [comment]
> I faced the same problem and initially built my own server-side connector to sync my CalDAV calendar with Microsoft 365 via the Graph API. The approach worked well enough that I turned it into a proper product: CalDAVconnect (https://caldavconnect.de/). Currently in public beta and free to use.
> 
> Instead of relying on a COM add-in inside Outlook, it syncs your CalDAV server (Nextcloud, SOGo, Radicale, etc.) directly with Microsoft 365 - server-to-server, in real time. It works with New Outlook, Classic Outlook, Outlook Web, and Outlook mobile, since the sync happens at the cloud level rather than the client level.

Great! Does this work with Zimbra community servers? Will it work with outlook.com calendars?

**Comment by schindlerchristian** [comment]
@cymumtaz CalDAVconnect works with any standard CalDAV server. Zimbra Community Edition supports CalDAV, so it should work - just select "Generic CalDAV" as the server type in the setup wizard. That said, we haven't tested Zimbra specifically yet. If you try it, I'd love to hear how it goes!

Outlook.com: CalDAVconnect supports both personal Microsoft accounts (Outlook.com, Hotmail, Live) and Microsoft 365 work/school accounts. You connect your CalDAV server on one side and your Outlook.com calendar on the other - everything syncs bidirectionally.

**Comment by Mannshoch** [comment]
I checked with my IT what their condition is, so I'm allowed to further use a sync to my private CalDAV.

As far, they have limited the use of application from outside the company to the web-client and the Official Outlook/Teams app. So I'm not allowed to access company data over Graph-API from outside.

It's a different case for inside the company. I'm allowed to install a client side application as I did for caldavsynchronizer.
Our IT prefer that solution because that way, they have also the used app on check with Defender.

Conclusion, I prefer a Windows Service application that runs on my work PC and sync to my CalDAV server.

**Comment by cymumtaz** [comment]
> [@cymumtaz](https://github.com/cymumtaz) CalDAVconnect works with any standard CalDAV server. Zimbra Community Edition supports CalDAV, so it should work - just select "Generic CalDAV" as the server type in the setup wizard. That said, we haven't tested Zimbra specifically yet. If you try it, I'd love to hear how it goes!
> 
> Outlook.com: CalDAVconnect supports both personal Microsoft accounts (Outlook.com, Hotmail, Live) and Microsoft 365 work/school accounts. You connect your CalDAV server on one side and your Outlook.com calendar on the other - everything syncs bidirectionally.

So the problem is I need to activate CalDav in order for this to work. I'll talk with my manager about this.

**Comment by coopee1-dell** [comment]
> [@schindlerchristian](https://github.com/schindlerchristian) while an interesting idea, it's unfortunately neither open source nor self-hostable, which is a deal breaker at least for me.
> 
> Nothing wrong with earning some money, but advertising a closed source, proprietary product/service on github may be the wrong approach.

My Org explicitly blocked that integration.
